packages/core/src/telemetry/clearcut-logger/clearcut-logger.ts

```diff
@@ -62,6 +66,19 @@ export class ClearcutLogger {
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Clearcut expects this format.
   enqueueLogEvent(event: any): void {
+    // Prevent unbounded memory growth by limiting event queue size
+    if (this.events.length >= this.max_events) {
+      // Remove oldest events (FIFO) to make room for new ones
+      const eventsToRemove = this.events.length - this.max_events + 1;
+      this.events.splice(0, eventsToRemove);
```

Comment:
same comment as bellow about useing
https://yomguithereal.github.io/mnemonist/fixed-deque
or a doubly linked list to avoid O(events.length) operations.